tests/unit/introspection.tcl

```diff
@@ -63,21 +71,23 @@ start_server {tags {"introspection"}} {
         set output3 [get_field_in_client_list $rd_id $info_list "tot-net-out"]
         set cmd3 [get_field_in_client_list $rd_id $info_list "tot-cmds"]
         $rd blpop mylist 0
-        set info_list [r client list]
-        set input4 [get_field_in_client_list $rd_id $info_list "tot-net-in"]
-        set output4 [get_field_in_client_list $rd_id $info_list "tot-net-out"]
-        set cmd4 [get_field_in_client_list $rd_id $info_list "tot-cmds"]
-        assert_equal [expr $input3+34] $input4
-        assert_equal $output3 $output4
-        assert_equal $cmd3 $cmd4
+        set input4 [expr $input3 + 34]
+        set output4 $output3
+        set cmd4 $cmd3
+        wait_for_condition 5 100 {
+            [list $input4 $output4 $cmd4] eq [get_client_tot_in_out_cmds $rd_id]
+        } else {
+            fail "Blocked BLPOP didn't only increment client's tot-net-in"
```

Comment:
let's also print the vars? I remember that wait_for_condition will not print these assert_equal contexts

```suggestion
            puts "input4: $input4, output4: $output4, $cmd4: cmd4"
            puts [get_client_tot_in_out_cmds $rd_id]
            fail "Blocked BLPOP didn't only increment client's tot-net-in"
```